src/OrbitGl/GraphTrack.cpp

```diff
@@ -291,30 +292,34 @@ void GraphTrack<Dimension>::DrawSeries(PrimitiveAssembler& primitive_assembler,
     auto next_it = std::next(current_it);
     uint64_t next_time = std::min(next_it->first, max_tick);
 
-    if (current_it == entries.begin()) {
-      aggr.StartNewEntry(current_time, next_time, normalized_cumulative_values);
+    if (aggr.GetAccumulatedEntry() == nullptr) {
+      aggr.SetEntry(current_time, next_time, normalized_cumulative_values);
     } else {
-      // If the size of the current entry is less than a pixel
-      if (aggr.GetEntry().end_tick < next_pixel_start_ns) {
+      // If the accumulated entry still fits in the pixel
+      if (aggr.GetAccumulatedEntry()->end_tick < next_pixel_start_ns) {
         // Add the current data to accumulated_entry
-        aggr.AppendData(current_time, next_time, normalized_cumulative_values);
+        aggr.MergeDataIntoEntry(current_time, next_time, normalized_cumulative_values);
       } else {
         // Otherwise, draw the accumulated_entry and start accumulating a new one
-        DrawSingleSeriesEntry(primitive_assembler, aggr.GetEntry().start_tick,
-                              aggr.GetEntry().end_tick, aggr.GetEntry().values, z);
+        // When drawing we only use max values - for every usage of this track
```

Comment:
From your description: 
> The graph track only draws boxes, so the largest box would overdraw the smaller ones anyway.

Please include something related to the boxes in the comment. It wasn't clear for me.